luigi/parameter.py

```diff
@@ -959,3 +960,37 @@ def serialize(self, x):
         :param x: the value to serialize.
         """
         return json.dumps(x)
+
+class NumericalParameter(Parameter):
+    """Parameter that restricts values to the given numerical type, i.e. int or
+    float, and to the specified range
+    """
+    def __init__(self, var_type=int, min_value=0, max_value=maxsize,
+                 left_op=operator.lt, right_op=operator.le, *args, **kwargs):
+        super(NumericalParameter, self).__init__(*args, **kwargs)
+        self.var_type = var_type
+        self.min_value = min_value
+        self.max_value = max_value
+        self.left_op = left_op
+        self.right_op = right_op
+        self.permitted_range = (
+            "{var_type} in {left_endpoint}{min_value}, {max_value}{right_endpoint}".format(
+                var_type=var_type.__name__,
+                min_value=min_value, max_value=max_value,
+                left_endpoint="(" if self.left_op == operator.lt else "[",
+                right_endpoint=")" if self.right_op == operator.lt else "]"))
+        if self.description:
+            self.description += " "
+        else:
+            self.description = ""
+        self.description += "permitted values: " + self.permitted_range
```

Comment:
Can you make all variables private? I guess the exception is `.description` as the parent class will use it, but there's not reason to expose these variables to users of this class.